libcaf_core/caf/policy/work_stealing.hpp

```diff
@@ -57,23 +57,32 @@ class work_stealing : public unprofiled {
 
   // The coordinator has only a counter for round-robin enqueue to its workers.
   struct coordinator_data {
-    std::atomic<size_t> next_worker;
-    inline coordinator_data() : next_worker(0) {
+    inline explicit coordinator_data(scheduler::abstract_coordinator*)
+        : next_worker(0) {
       // nop
     }
+
+    std::atomic<size_t> next_worker;
   };
 
   // Holds job job queue of a worker and a random number generator.
   struct worker_data {
+    inline explicit worker_data(scheduler::abstract_coordinator* p)
+          // when `p->num_workers() < 2`,
+          // this saves us from the `std::random_device{}()` call
+        : rengine(p->num_workers() < 2 ? 0 : std::random_device{}()),
```

Comment:
This micro optimization isn't useful. The few cycles we "waste" when running with a single worker isn't worth the code IMO.